coalib/bearlib/naming_conventions/__init__.py

```diff
@@ -73,6 +73,8 @@ def to_snakecase(string):
     'already_snake_case'
     >>> to_snakecase('   string  ')
     '___string__'
+    >>> to_snakecase('Hello.ThisWorld')
+    'hello.this_world'
```

Comment:
sorry but I feel like you're not really solving the issue here but just "make it work with the tests" - now `HelloThis.World` won't be translated correctly I think.